pkg/image/importer/importer.go

```diff
@@ -129,6 +132,11 @@ func (i *ImageStreamImporter) importImages(ctx gocontext.Context, retriever Repo
 		repoName := defaultRef.RepositoryName()
 		registryURL := defaultRef.RegistryURL()
 
+		if !importer.isAllowedRegistry(defaultRef.RegistryHostname()) {
+			isi.Status.Images[i].Status = invalidStatus("", field.Invalid(field.NewPath("from", "name"), from.Name, fmt.Sprintf("registry %q not allowed in master configuration", defaultRef.RegistryHostname())))
```

Comment:
with the current implementation, it will get surfaced in the events inside image stream... when I rework this in validation, I think it you will get the error when you run `oc import-image`, so the IS won't be created at all.

Web console will get validation error as well, so I think that will be presented to users in nice form.